vertical-pod-autoscaler/pkg/recommender/model/aggregations_config.go

```diff
@@ -17,22 +17,23 @@ limitations under the License.
 package model
 
 import (
+	"flag"
 	"time"
 
 	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/util"
 )
 
 var (
 	// MemoryAggregationWindowLength is the length of the memory usage history
-	// aggregated by VPA, which is 8 days.
-	MemoryAggregationWindowLength = time.Hour * 8 * 24
+	// aggregated by VPA.
+	MemoryAggregationWindowLength = DefaultMemoryAggregationWindowLength
 	// MemoryAggregationInterval is the length of a single interval, for
 	// which the peak memory usage is computed.
 	// Memory usage peaks are aggregated in daily intervals. In other words
 	// there is one memory usage sample per day (the maximum usage over that
 	// day).
 	// Note: AggregationWindowLength must be integrally divisible by this value.
```

Comment:
There are configurations which will be invalid with the current algorithm. Perhaps we should consider exposing different parameters as flags - for example MemoryAggregationInterval and MemoryAggregationWindowIntervalCount. Then MemoryAggregationWindow = MemoryAggregationInterval *  MemoryAggregationWindowIntervalCount

Reply:
Is this to make sure that `MemoryAggregationWindow` would be integrally divisible by `MemoryAggregationInterval`?

Reply:
I have changed the flag to [`memory-aggregation-window-interval-count`](https://github.com/kubernetes/autoscaler/pull/2905/files#diff-483c01ba6f34fd6f8c9d4944684cff98R55) and introduced a [func](https://github.com/kubernetes/autoscaler/pull/2905/files#diff-483c01ba6f34fd6f8c9d4944684cff98R55) to compute `MemoryAggregationWindowLength`.